app/controllers/admin/buildings_controller.rb

```diff
@@ -28,22 +28,44 @@ def update
       end
     end
 
-    private
-
-      def building_params
-        params.require(:building).permit(
-          :uprn,
-          :building_name,
-          :street,
-          :postcode,
-          :land_registry_proprietor_address,
-          :land_registry_proprietor_name,
-          :proprietor_email,
-        )
+    def bulk_update
+      if params[:building] && Building.find(params[:building][:building_id].map(&:to_i))
```

Comment:
`building_params` already `.require(:building)` so I don't think I'd double check here.

Reply:
could we move the logic to the model? so have a class method that updates a collection of records, and then the controller can stick to the usual `call the model -> set the flash -> redirect`.

Reply:
also you might want to use `Building.where(id: [the array of ids]).update_all(on_delta: 1)` to generate 2 queries rather than N x 2.

Reply:
Have done this - it works much better.